ts/Stock/Indicators/DisparityIndex/DisparityIndexIndicator.ts

```diff
@@ -0,0 +1,286 @@
+
+/**
+ * The Disparity Index series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.disparityindex
+ *
+ * @augments Highcharts.Series
+ */
+class DisparityIndexIndicator extends SMAIndicator {
+    /**
+     * Disparity Index.
+     * This series requires the `linkedTo` option to be set and should
+     * be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/disparity-index
+     *         Disparity Index indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/disparity-index
+     * @optionparent plotOptions.disparityindex
+     */
+    public static defaultOptions: DisparityIndexOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            /**
+             * The average used to calculate the Disparity Index indicator.
+             * By default it uses SMA. To use other averages, e.g. EMA,
+             * the stock/indicators/ema.js file needs to be loaded.
+             *
+             * If value is different than ema|dema|tema|wma, then sma is used.
+             */
+            average: 'sma',
+            index: 3,
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    } as DisparityIndexOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<DisparityIndexPoint> = void 0 as any;
+    public options: DisparityIndexOptions = void 0 as any;
+    public points: Array<DisparityIndexPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(this: DisparityIndexIndicator): void {
+        const args = arguments,
+            ctx = this, // Disparity Index indicator
+            params = args[1].params, // options.params
+            averageType = params && params.average ? params.average : void 0;
+
+        let averageIndicator;
+
+        switch (averageType) {
+        case 'ema':
+            averageIndicator = EMAIndicator;
+            break;
+        case 'dema':
+            averageIndicator = DEMAIndicator;
+            break;
+        case 'tema':
+            averageIndicator = TEMAIndicator;
+            break;
+        case 'wma':
+            averageIndicator = WMAIndicator;
+            break;
+        default: // use sma if any of the above strings do not match
+            averageIndicator = SMAIndicator;
+        }
+
+        ctx.averageIndicator = averageIndicator;
+
+        extend(DisparityIndexIndicator.prototype, {
+            nameBase: 'Disparity Index (' + averageType + ')'
```

Comment:
Use `nameComponents`, so the resulting format will match other technical indicators:
` Name (paramA, paramB, ... , paramC)` => `Disparity Index (14, EMA)`